File: frontend/src/api/games.ts
import { http } from './client'
import type { CreateGameRequest, GameDTO } from '../types/api'

const games = (sessionId: string) => `/games/${sessionId}`

export const createGame = (data: CreateGameRequest = {}) =>
  http.post<GameDTO>('/games', data).then((r) => r.data)

export const getGame = (sessionId: string) =>
  http.get<GameDTO>(games(sessionId)).then((r) => r.data)

export const askQuestion = (sessionId: string, text: string) =>
  http.post<GameDTO>(`${games(sessionId)}/questions`, { text }).then((r) => r.data)

export const makeGuess = (sessionId: string, text: string) =>
  http.post<GameDTO>(`${games(sessionId)}/guesses`, { text }).then((r) => r.data)

export const skipPhase = (
  sessionId: string,
  currentPhase: 'question' | 'guess',
) =>
  http
    .post<GameDTO>(`${games(sessionId)}/skip`, { current_phase: currentPhase })
    .then((r) => r.data)

export const retry = (sessionId: string) =>
  http.post<GameDTO>(`${games(sessionId)}/retry`, {}).then((r) => r.data)

export const abandon = (sessionId: string) =>
  http.post<GameDTO>(`${games(sessionId)}/abandon`, {}).then((r) => r.data)

File: backend/app/api/routes/games.py
"""/api/v1/games 路由：仅做入参解析 → 调 orchestrator → 返回 DTO。"""

from __future__ import annotations

from fastapi import APIRouter

from ...schemas.game import (
    AskRequest,
    CreateGameRequest,
    GameDTO,
    GuessRequest,
    SkipRequest,
)
from ...services import orchestrator

router = APIRouter(prefix="/games", tags=["games"])


@router.post("", response_model=GameDTO, status_code=201)
async def create_game(req: CreateGameRequest) -> GameDTO:
    return await orchestrator.create_game(req)


@router.get("/{session_id}", response_model=GameDTO)
async def get_game(session_id: str) -> GameDTO:
    return await orchestrator.get_game(session_id)


@router.post("/{session_id}/questions", response_model=GameDTO)
async def ask_question(session_id: str, req: AskRequest) -> GameDTO:
    return await orchestrator.ask(session_id, req.text)


@router.post("/{session_id}/guesses", response_model=GameDTO)
async def make_guess(session_id: str, req: GuessRequest) -> GameDTO:
    return await orchestrator.guess(session_id, req.text)


@router.post("/{session_id}/skip", response_model=GameDTO)
async def skip_phase(session_id: str, req: SkipRequest) -> GameDTO:
    return await orchestrator.skip(session_id, req.current_phase)


@router.post("/{session_id}/retry", response_model=GameDTO)
async def retry(session_id: str) -> GameDTO:
    return await orchestrator.retry(session_id)


@router.post("/{session_id}/abandon", response_model=GameDTO)
async def abandon(session_id: str) -> GameDTO:
    return await orchestrator.abandon(session_id)
